e3sm_diags/derivations/acme.py

```diff
@@ -654,9 +665,18 @@ def cosp_histogram_standardize(cld: "FileVariable"):
                 ("prw",),
                 lambda prw: convert_units(rename(prw), target_units="kg/m2"),
             ),
+            (
+                ("VapWaterPath",),  # EAMxx
+                lambda prw: convert_units(rename(prw), target_units="kg/m2"),
+            ),
+        ]
+    ),
+    "SOLIN": OrderedDict(
+        [
+            (("rsdt",), rename),
+            (("SW_flux_dn_at_model_top",), rename),  # EAMxx
         ]
```

Comment:
```suggestion
    "SOLIN": {
         ("rsdt",): rename
         ("SW_flux_dn_at_model_top",): rename,  # EAMxx
      }
```

I don't think you need need to use an `OrderedDict` in the nested derived variables dictionary. I just use a regular dictionary in my refactored version (`derivations.py`). The old Dataset class will probably be able to parse this the same way as the new Dataset class in the CDAT migration branch because order of the keys shouldn't matter.

https://github.com/E3SM-Project/e3sm_diags/blob/6b296b1fc4f8ddaea2a5efaa6639b95a31a009a8/e3sm_diags/derivations/derivations.py#L592-L604.

Reply:
I recognize you're updating existing elements that are already OrderedDict. Up to you if you want to use a regular dict for new or existing elements that you've modified. Just a suggestion since it is easier to read and cleaner, but I'm aware this file is going to be replaced anyways.

Reply:
Hi @tomvothecoder Thank you for reviewing. It's been a while since the ordered dictionary was designed. I think we made it ordered dictionary to support edge cases when some datasets include multiple variables in the key, and we wanted to loop over keys in order. I'm not sure if this is still needed given that lots of datasets are being updated over years.

Reply:
Maybe e3sm_diags development started around Python 3.5/3.6? I learned that the regular dict will maintain order with Python >=3.6, so we probably don't need to use `OrderedDict` anymore.

> Starting with CPython 3.6, dictionaries [return items in the order you inserted them](https://docs.python.org/3/whatsnew/3.6.html#new-dict-implementation).
> Source - https://stackoverflow.com/a/47849121

> Since Python 3.7, all dictionaries are guaranteed to be ordered. The Python contributors determined that switching to making dict ordered would not have a negative performance impact. I don't know how the performance of OrderedDict compares to dict in Python >= 3.7, but I imagine they would be comparable since they are both ordered.
> 
> Note that there are still differences between the behaviour of OrderedDict and dict. See also: [Will OrderedDict become redundant in Python 3.7](https://stackoverflow.com/q/50872498/247696)?
> Source - https://stackoverflow.com/a/53535866

Reply:
Yeah, i think the development started around 3.5 and 3.6. In this case we should go with regular dictionary. Good call for the cdat migration!

Reply:
Maybe `e3sm_diags` was developed around Python 3.5/3.6?

I learned that Python >=3.6 maintains order for regular dictionaries and that `OrderedDict` is probably not needed for our use-case anymore (get keys in order for matching in the dataset).

> Are dictionaries ordered in Python 3.6+?
> They are insertion ordered[1].
> 
> As of Python 3.6, for the CPython implementation of Python, dictionaries remember the order of items inserted. This is considered an implementation detail in Python 3.6; you need to use OrderedDict if you want insertion ordering that's guaranteed across other implementations of Python (and other ordered behavior[1]).
> 
> As of Python 3.7, this is a guaranteed language feature, not merely an implementation detail. [From a python-dev message by GvR](https://mail.python.org/pipermail/python-dev/2017-December/151283.html):
> Source: https://stackoverflow.com/a/39980744

Reply:
No problem!